File: sdk/go/provider/ta_easy.go
package provider

import (
	"bytes"
	"encoding/json"
	"fmt"
	"io"

	http "github.com/bogdanfinn/fhttp"
)

const taEasyAPIBase = "https://api-endpoint.ta-easy.com"
const taEasyOrigin = "https://www.ta-easy.com"

func setTaEasyHeaders(req *http.Request) {
	req.Header.Set("Accept", "application/json")
	req.Header.Set("User-Agent", GetCurrentUA())
	req.Header.Set("Origin", taEasyOrigin)
	req.Header.Set("Referer", taEasyOrigin+"/")
}

// TaEasyGenerate 创建临时邮箱
// API: POST /temp-email/address/new（空 body）
func TaEasyGenerate() (*CreatedMailbox, error) {
	req, err := http.NewRequest("POST", taEasyAPIBase+"/temp-email/address/new", nil)
	if err != nil {
		return nil, err
	}
	setTaEasyHeaders(req)
	req.Header.Set("Content-Length", "0")

	client := HTTPClient()
	resp, err := client.Do(req)
	if err != nil {
		return nil, err
	}
	defer resp.Body.Close()
	if err := CheckHTTPStatus(resp, "ta-easy generate"); err != nil {
		return nil, err
	}
	body, err := io.ReadAll(resp.Body)
	if err != nil {
		return nil, err
	}
	var out struct {
		Status    string `json:"status"`
		Message   string `json:"message"`
		Address   string `json:"address"`
		Token     string `json:"token"`
		ExpiresAt int64  `json:"expiresAt"`
	}
	if err := json.Unmarshal(body, &out); err != nil {
		return nil, err
	}
	if out.Status != "success" || out.Address == "" || out.Token == "" {
		if out.Message != "" {
			return nil, fmt.Errorf("ta-easy: %s", out.Message)
		}
		return nil, fmt.Errorf("ta-easy: create failed")
	}
	return &CreatedMailbox{
		Channel:   "ta-easy",
		Email:     out.Address,
		Token:     out.Token,
		ExpiresAt: out.ExpiresAt,
	}, nil
}

// TaEasyGetEmails 拉取收件箱
// API: POST /temp-email/inbox/list，JSON body: { "token", "email" }
func TaEasyGetEmails(email, token string) ([]NormEmail, error) {
	payload, err := json.Marshal(map[string]string{
		"token": token,
		"email": email,
	})
	if err != nil {
		return nil, err
	}
	req, err := http.NewRequest("POST", taEasyAPIBase+"/temp-email/inbox/list", bytes.NewReader(payload))
	if err != nil {
		return nil, err
	}
	setTaEasyHeaders(req)
	req.Header.Set("Content-Type", "application/json")

	client := HTTPClient()
	resp, err := client.Do(req)
	if err != nil {
		return nil, err
	}
	defer resp.Body.Close()
	if err := CheckHTTPStatus(resp, "ta-easy inbox"); err != nil {
		return nil, err
	}
	raw, err := io.ReadAll(resp.Body)
	if err != nil {
		return nil, err
	}
	var wrap struct {
		Status  string            `json:"status"`
		Message string            `json:"message"`
		Data    []json.RawMessage `json:"data"`
	}
	if err := json.Unmarshal(raw, &wrap); err != nil {
		return nil, err
	}
	if wrap.Status != "success" {
		if wrap.Message != "" {
			return nil, fmt.Errorf("ta-easy: %s", wrap.Message)
		}
		return nil, fmt.Errorf("ta-easy: inbox failed")
	}
	out := make([]NormEmail, 0, len(wrap.Data))
	for _, rm := range wrap.Data {
		var m map[string]interface{}
		if err := json.Unmarshal(rm, &m); err != nil {
			continue
		}
		out = append(out, NormalizeMap(m, email))
	}
	return out, nil
}

File: sdk/npm/src/providers/ta-easy.ts
import { InternalEmailInfo, Email, Channel } from '../types';
import { normalizeEmail } from '../normalize';
import { fetchWithTimeout } from '../retry';

const CHANNEL: Channel = 'ta-easy';
const API_BASE = 'https://api-endpoint.ta-easy.com';
const ORIGIN = 'https://www.ta-easy.com';

function browserHeaders(): Record<string, string> {
  return {
    Accept: 'application/json',
    'User-Agent':
      'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/146.0.0.0 Safari/537.36 Edg/146.0.0.0',
    origin: ORIGIN,
    referer: `${ORIGIN}/`,
  };
}

/** POST /temp-email/address/new（空 body） */
export async function generateEmail(): Promise<InternalEmailInfo> {
  const res = await fetchWithTimeout(`${API_BASE}/temp-email/address/new`, {
    method: 'POST',
    headers: { ...browserHeaders(), 'Content-Length': '0' },
  });
  if (!res.ok) {
    throw new Error(`ta-easy generate: ${res.status}`);
  }
  const data = await res.json();
  if (data.status !== 'success' || !data.address || !data.token) {
    throw new Error(data.message ? `ta-easy: ${data.message}` : 'ta-easy: create failed');
  }
  return {
    channel: CHANNEL,
    email: data.address as string,
    token: data.token as string,
    expiresAt: typeof data.expiresAt === 'number' ? data.expiresAt : undefined,
  };
}

/** POST /temp-email/inbox/list */
export async function getEmails(email: string, token: string): Promise<Email[]> {
  const res = await fetchWithTimeout(`${API_BASE}/temp-email/inbox/list`, {
    method: 'POST',
    headers: { ...browserHeaders(), 'Content-Type': 'application/json' },
    body: JSON.stringify({ token, email }),
  });
  if (!res.ok) {
    throw new Error(`ta-easy inbox: ${res.status}`);
  }
  const data = await res.json();
  if (data.status !== 'success') {
    throw new Error(data.message ? `ta-easy: ${data.message}` : 'ta-easy: inbox failed');
  }
  const list = Array.isArray(data.data) ? data.data : [];
  return list.map((raw: any) => normalizeEmail(raw, email));
}
